File: web/src/services/plan/PlanService.js
import { request,download,   METHOD } from "@/utils/request";
import Cookies from "js-cookie";
export function buildGantt(projectId) {
  return request({
    url: "/tracker/plan/gantt",
    method: METHOD.GET,
    params: { projectId: projectId },
  });
}

export function findPlans(projectId, keyword) {
  return request({
    url: "/tracker/plan/list",
    method: METHOD.GET,
    params: { projectId: projectId, keyword: keyword },
  });
}

export function autoPlan(projectId) {
  return request({
    url: "/tracker/plan/auto",
    method: METHOD.POST,
    params: { projectId: projectId },
  });
}

export function findMilestones(projectId) {
  return request({
    url: "/tracker/plan/milestones",
    method: METHOD.GET,
    params: { projectId: projectId },
  });
}

export function findWaitExecutePlans(projectId) {
  return request({
    url: "/tracker/plan/waitExecute",
    method: METHOD.GET,
    params: { projectId: projectId },
  });
}

export function findOnePlan(testRunId) {
  return request({
    url: `/tracker/plan/${testRunId}`,
    method: METHOD.GET,
  });
}
export function createPlan(parameter) {
  return request({
    url: "/tracker/plan",
    method: METHOD.POST,
    data: parameter,
  });
}

export function updatePlan(parameter) {
  return request({
    url: "/tracker/plan",
    method: METHOD.PUT,
    data: parameter,
  });
}
export function deletePlan(id) {
  return request({
    url: "/tracker/plan/" + id,
    method: METHOD.DELETE,
  });
}
export function updateDeliverables(testRunId, data) {
  return request({
    url: `/tracker/plan/${testRunId}/deliverables`,
    method: METHOD.POST,
    data: data,
  });
}

export function uploadAttachment(deliverableId, attachment) {
  Object.assign(attachment, { objectId: deliverableId, type: "ATTACHMENT" });
  return request({
    url: `/tracker/plan/deliverable/${deliverableId}/attachments`,
    method: METHOD.POST,
    data: attachment,
  });
}

export function updateDeliverable(deliverableId, data) {
  return request({
    url: `/tracker/plan/deliverable/${deliverableId}`,
    method: METHOD.POST,
    data: data,
  });
}

export function findDeliverables(projectId) {
  return request({
    url: `/tracker/plan/deliverable/list`,
    method: METHOD.GET,
    params: { projectId: projectId },
  });
}

export function associateSprints(testRunId, sprints) {
  return request({
    url: `/tracker/plan//${testRunId}/associateSprints`,
    method: METHOD.POST,
    data: sprints,
  });
}

export function dissociateSprints(testRunId, sprints) {
  return request({
    url: `/tracker/plan//${testRunId}/dissociateSprints`,
    method: METHOD.POST,
    data: sprints,
  });
}

export function associateTrackerItems(testRunId, trackerItems) {
  return request({
    url: `/tracker/plan//${testRunId}/associateTrackerItems`,
    method: METHOD.POST,
    data: trackerItems,
  });
}

export function dissociateTrackerItems(testRunId, trackerItems) {
  return request({
    url: `/tracker/plan//${testRunId}/dissociateTrackerItems`,
    method: METHOD.POST,
    data: trackerItems,
  });
}

export function changePlanOrders(plans) {
  return request({
    url: `/tracker/plan/changeOrder`,
    method: METHOD.POST,
    data: plans,
  });
}

export function importMPP(projectId, file) {
  return request({
    url: `/tracker/plan/import-ms-project`,
    method: METHOD.POST,
    params: { projectId: projectId, file: file },
  });
}

export function exportPlans(projectId) {

  let url =`/tracker/plan/export`;

  download(url, {projectId:projectId});
}

File: server/helm-tracker/src/main/java/com/harbortek/helm/tracker/service/PlanService.java
/*
 * Copyright [2025] [Harbortek Corp.]
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.harbortek.helm.tracker.service;

import com.harbortek.helm.tracker.vo.items.TrackerItemVo;
import com.harbortek.helm.tracker.vo.log.AttachmentVo;
import com.harbortek.helm.tracker.vo.plan.DeliverableVo;
import com.harbortek.helm.tracker.vo.plan.GanttVo;
import com.harbortek.helm.tracker.vo.plan.PlanVo;
import com.harbortek.helm.tracker.vo.plan.SprintVo;

import java.io.InputStream;
import java.util.Collection;
import java.util.List;

public interface PlanService {

    GanttVo buildGantt(Long projectId);
    Collection<PlanVo> findPlans(Long projectId);

    Collection<PlanVo> findMilestones(Long projectId);

    PlanVo findOnePlan(Long id);

    PlanVo createPlan(PlanVo planVo);

    PlanVo updatePlan(PlanVo planVo);

    void deletePlan(Long id);

    void updateDeliverables(Long planId, List<DeliverableVo> deliverables);

    AttachmentVo uploadAttachment(Long deliverableId, AttachmentVo attachmentVo);

    void updateDeliverable(Long deliverableId, DeliverableVo deliverableVo);

    void autoPlans(Long projectId);

    Collection<DeliverableVo> findDeliverables(Long projectId);

    Collection<PlanVo> findWaitExecutePlans(Long projectId, Long currentUserId);

    void associateSprints(Long planId, List<SprintVo> sprints);

    void associateTrackerItems(Long planId, List<TrackerItemVo> trackerItems);

    void dissociateSprints(Long planId, List<SprintVo> sprints);

    void dissociateTrackerItems(Long planId, List<TrackerItemVo> trackerItems);

    void changeOrder(List<PlanVo> planVos);

    void importMPP(Long projectId, InputStream mppFile);
}
